### PR TITLE
rpm-packaging: Add rhsm links

### PR DESCRIPTION
**Pull request by cgwalters**
This pulls in the code from https://github.com/openshift/os/pull/876/commits/dd35dd0e102aeed90df14f05c8ae9da4c8c5962a for the same reason.

### CODE REVIEW

**Comment by shi2wei3** [comment]
what about the redhat.repo? I still cannot install pkgs

# dnf install kernel
Updating Subscription Management repositories.
Unable to read consumer identity
subscription-manager is operating in container mode.
Error: There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".

**Comment by cgwalters** [comment]
I've verified this one works btw.

File: tier-1/rpm-packaging.yaml [hunk 1]
```diff
@@ -4,3 +4,13 @@ packages:
   - dnf
   # To ensure we can enable client certs to access RHEL content
   - subscription-manager
+
+postprocess:
+  # These enable librhsm which enables host subscriptions to work in containers
+  # https://github.com/rpm-software-management/librhsm/blob/fcd972cbe7c8a3907ba9f091cd082b1090231492/rhsm/rhsm-context.c#L30
+  # https://github.com/openshift/os/pull/876/commits/dd35dd0e102aeed90df14f05c8ae9da4c8c5962a
+  - |
+    #!/usr/bin/bash
```
Comment (dcantrell) [inline tier-1/rpm-packaging.yaml:13]: My only comment here is to not use #!/usr/bin/bash when nothing is bash-specific.  bash is not as portable as many think and bash as shipped by Apple still is at 3.2.57 due to GPLv3 (not really hugely relevant, it's just that a large installation of bash out in the wild is 3.2.57).  At any rate, this is just my own personal nitpick.  Just say #!/bin/sh or #!/usr/bin/sh and stick with portable shell syntax when possible.  Yes, we still have to deal with that in 2024.

**Review by sallyom — APPROVED** [review]
LGTM (other than the bash/sh comment but I'm fine with bash or sh)

File: tier-1/rpm-packaging.yaml (hunk 1 above)
Comment (cgwalters) [reply tier-1/rpm-packaging.yaml:13]: True...though in practice we know we're executing the code in this container image which has the bash we want.

Ultimately this specific bit and others will get replaced with a plain `RUN` invocation in a container build which has that same semantic.

**Pull request merged**